lib/Mojo/Base.pm

```diff
@@ -72,7 +72,6 @@ sub import {
   # Role
   elsif ($flags[0] eq '-role') {
     Carp::croak 'Role::Tiny 2.000001+ is required for roles' unless ROLES;
-    eval "package $caller; use Role::Tiny; 1" or die $@;
   }
```

Comment:
Think this block should go away and the croak should be moved down into the new if -roles check below.